File: 프로그래머스/알고리즘 고득점 Kit/깊이,너비 우선 탐색(DFS,BFS)/Level 3/네트워크/네트워크.py
def solution(n, computers):
    visit = [False for i in range(n)]

    cnt = 0
    for i in range(n):
        if visit[i]: continue

        q = [i]
        visit[i] = True
        while len(q) > 0:
            tmp = q[0]
            q.pop(0)

            for j in range(n):
                if not visit[j] and computers[tmp][j]:
                    q.append(j)
                    visit[j] = True

        cnt += 1

    return cnt

File: 프로그래머스/알고리즘 고득점 Kit/깊이,너비 우선 탐색(DFS,BFS)/Level 3/네트워크/네트워크.cpp
#include <string>
#include <vector>
#include <queue>
using namespace std;

int solution(int n, vector<vector<int>> computers) {
    vector<bool> visit(n, false);
    
    int cnt = 0;
    for(int i=0; i<n; i++){
        if(visit[i]) continue;
        
        queue<int> q;
        q.push(i);
        visit[i] = true;
        while(!q.empty()){
            int tmp = q.front();
            q.pop();
            
            for(int j=0; j<n; j++){
                if(!visit[j] && computers[tmp][j]){
                    q.push(j);
                    visit[j] = true;
                }
            }
        }
        cnt++;
    }
    
    return cnt;
}
